presto-hive/src/main/java/com/facebook/presto/hive/OrcFileWriterFactory.java

```diff
@@ -124,7 +123,7 @@ public OrcFileWriterFactory(
             NodeVersion nodeVersion,
             DateTimeZone hiveStorageTimeZone,
             FileFormatDataSourceStats readStats,
-            OrcWriterOptions orcWriterOptions,
+            OrcFileWriterConfig orcFileWriterConfig,
```

Comment:
OrcFileWriterConfig is mutable. By replacing OrcWriterOptions with OrcFileWriterConfig you exposed this class to changes that might be made to OrcFileWriterConfig after passing it into the constructor and actually using in down below in the createFileWriter method.

I don't know about the lifecycle of OrcFileWriterConfig in Presto, but I'd not replace immutable object with a mutable version unless I have a good reasoning for it.

Reply:
This is a great point. Today OrcFileWriterConfig is used for config serialization and de-serialization and once deserialized it is immutable. So the current change is ok. 

But to future proof the code and to maintain the semantics, instead of capturing the OrcFileWriterConfig , I will capture the OrcWriterOptions.Builder which will provide the same level of isolation as the existing code.

Reply:
Are you sure? From my understanding of this change you pass the OrcFileWriterConfig to be able to get the builder later in the `createFileWriter`. If you add a new method `OrcWriterOptions.toBuilder`, that I suggested previously, you will reduce amount of changes in this class and keep exactly the same behavior as it was before this change.

Opinion?

Everything else look good to me.

Reply:
I'm sorry, you changed it in the latest commit. LGTM

Reply:
Discussed offline. The Product code uses Injected constructor which tales in OrcFileWriterConfig. Previously this factory captured the OrcWriterOptions. Now it is capturing the OrcWriterOptions.Builder. 

Previous code, created immutable copies for each with* calls, The new code creates a copy of the builder, modifies all the properties and finally calls build.